File: scripts/download-work-images.py
"""Download case study images from algobrainai.com into public/work/."""
import os
import urllib.request
from urllib.parse import quote

UA = {"User-Agent": "Mozilla/5.0 (compatible; TRYN-image-fetch/1.0)"}

SLUG_IMAGES = {
    "whatsapp-healthcare": "/images/whatsapp bot in healthcare institute.png",
    "3d-product-chatbot": "/images/Gemini_Generated_Image_53cqja53cqja53cq.png",
    "excel-education": "/images/Gemini_Generated_Image_1bnki51bnki51bnk.png",
    "virtual-yoga": "/images/Gemini_Generated_Image_ekqtn6ekqtn6ekqt.png",
    "social-media-ai": "/images/Gemini_Generated_Image_uxl5s5uxl5s5uxl5.png",
    "diamond-analytics": "/images/diamond image.png",
    "voice-knowledge": "/images/voice-enabled-chatbot.jpg",
    "saree-draping": "/images/saree draping.png",
    "roleplay-wellness": "/images/wellness.jpg",
    "whatsapp-sales": "/images/ai_sales.png",
}

OUT_DIR = os.path.join(os.path.dirname(__file__), "..", "public", "work")


def download_image(img_path: str, dest: str) -> None:
    img_path = img_path.strip()
    url = "https://algobrainai.com" + quote(img_path)
    req = urllib.request.Request(url, headers=UA)
    with urllib.request.urlopen(req, timeout=30) as resp:
        data = resp.read()
    with open(dest, "wb") as f:
        f.write(data)


def main() -> None:
    os.makedirs(OUT_DIR, exist_ok=True)
    for slug, img_path in SLUG_IMAGES.items():
        ext = os.path.splitext(img_path.strip())[1].lower()
        dest = os.path.join(OUT_DIR, slug + ext)
        download_image(img_path, dest)
        print(f"/work/{slug}{ext}")


if __name__ == "__main__":
    main()

File: scripts/download-work-images.mjs
/**
 * Download industry-themed illustrative images for client case studies.
 * Run: node scripts/download-work-images.mjs
 */
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const OUT_DIR = path.join(__dirname, "..", "public", "work");

/** Curated Unsplash photo IDs by industry theme (illustrative only). */
const SLUG_IMAGES = {
  "whatsapp-healthcare":
    "https://images.unsplash.com/photo-1576091160399-112ba8d25d1d?w=1200&h=750&fit=crop",
  "3d-product-chatbot":
    "https://images.unsplash.com/photo-1617791160505-6f00504e3519?w=1200&h=750&fit=crop",
  "excel-education":
    "https://images.unsplash.com/photo-1523240795612-9a054b0db644?w=1200&h=750&fit=crop",
  "virtual-yoga":
    "https://images.unsplash.com/photo-1544367567-0f2fcb009e0b?w=1200&h=750&fit=crop",
  "social-media-ai":
    "https://images.unsplash.com/photo-1611162616305-c69b3fa7fbe0?w=1200&h=750&fit=crop",
  "diamond-analytics":
    "https://images.unsplash.com/photo-1581091226825-a6a2a5aee158?w=1200&h=750&fit=crop",
  "voice-knowledge":
    "https://images.unsplash.com/photo-1589254065878-42c9da997008?w=1200&h=750&fit=crop",
  "saree-draping":
    "https://images.unsplash.com/photo-1490481651871-ab68de25d43d?w=1200&h=750&fit=crop",
  "roleplay-wellness":
    "https://images.unsplash.com/photo-1506126613408-eca07ce68773?w=1200&h=750&fit=crop",
  "whatsapp-sales":
    "https://images.unsplash.com/photo-1556742049-0cfed4f6a45d?w=1200&h=750&fit=crop",
};

async function main() {
  fs.mkdirSync(OUT_DIR, { recursive: true });
  for (const [slug, url] of Object.entries(SLUG_IMAGES)) {
    const dest = path.join(OUT_DIR, `${slug}.jpg`);
    const res = await fetch(url);
    if (!res.ok) throw new Error(`Failed ${slug}: ${res.status}`);
    const buf = Buffer.from(await res.arrayBuffer());
    fs.writeFileSync(dest, buf);
    console.log(`saved ${dest}`);
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
